Specifications/QIR/Callables.md

```diff
@@ -205,9 +217,80 @@ The following snippet of LLVM code could be generated:
   [4 x void (%TupleHeader*, %TupleHeader*, %TupleHeader*)*]* @someOp,
   %TupleHeader* null)
 %g = call %__quantum__rt__callable_copy(%f)
-call %__quantum__rt__callable_adjoint(%g)
+call %__quantum__rt__callable_make_adjoint(%g)
 ```
 
+The actual implementation of the `%Callable` needs to support the following
+behavior:
+
+- For `__quantum__rt__callable_make_adjoint`, the parity of the number of
+  times this function has been applied should be tracked. The `Adjoint` functor
+  is its own inverse, so applying it twice is the same as not applying it at all.
+  Applying this function to a `%Callable` whose implementation table has a null
+  "adj" entry should cause a runtime failure.
+- For `__quantum__rt__callable_make_control`, the count of the number of 
+  times the function has been applied must be tracked. The `Controlled` functor
```

Comment:
```suggestion
  times the function has been applied must be tracked, since each application adds an additional item to the argument (explained in more detail below). The `Controlled` functor
```